From a1a0822e73f4f95ee2cc098febbc1494ad87101f Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Wed, 17 Feb 2021 13:46:40 -0700
Subject: [PATCH 13/15] DO-NOT-UPSTREAM: Various fixes/hacks/warning hiding

* drivers/input/joystick/analog.c: Avoid this warning on s390.

* drivers/net/ethernet/marvell/mvpp2/mvpp2.h: Avoid
-Wconstant-conversion warning (https://github.com/ClangBuiltLinux/linux/issues/997).

* lib/Kconfig.debug: Disable -Wframe-larger-than=. Getting all of these
fixed is a long term project.

* scripts/checksyscalls.sh: I do not care for this warning.

Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 drivers/input/joystick/analog.c            | 1 -
 drivers/net/ethernet/marvell/mvpp2/mvpp2.h | 2 +-
 lib/Kconfig.debug                          | 3 ++-
 scripts/checksyscalls.sh                   | 2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/input/joystick/analog.c b/drivers/input/joystick/analog.c
index 2b625ebef914..3935f8c95eef 100644
--- a/drivers/input/joystick/analog.c
+++ b/drivers/input/joystick/analog.c
@@ -157,7 +157,6 @@ static unsigned long analog_faketime = 0;
 #define GET_TIME(x)     do { x = analog_faketime++; } while(0)
 #define DELTA(x,y)	((y)-(x))
 #define TIME_NAME	"Unreliable"
-#warning Precise timer not defined for this architecture.
 #endif
 
 static inline u64 get_time(void)
diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2.h b/drivers/net/ethernet/marvell/mvpp2/mvpp2.h
index 834775843067..323e1c061f1d 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2.h
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2.h
@@ -748,7 +748,7 @@
 
 /* RX buffer constants */
 #define MVPP2_SKB_SHINFO_SIZE \
-	SKB_DATA_ALIGN(sizeof(struct skb_shared_info))
+	SKB_DATA_ALIGN(((int)(sizeof(struct skb_shared_info))))
 
 #define MVPP2_RX_PKT_SIZE(mtu) \
 	ALIGN((mtu) + MVPP2_MH_SIZE + MVPP2_VLAN_TAG_LEN + \
diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index dcf4a9028e16..fe52baf6dbf0 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -295,8 +295,9 @@ config ENABLE_MUST_CHECK
 	  attribute warn_unused_result" messages.
 
 config FRAME_WARN
-	int "Warn for stack frames larger than"
+	int "Warn for stack frames larger than" if !CC_IS_CLANG
 	range 0 8192
+	default 0 if CC_IS_CLANG
 	default 2048 if GCC_PLUGIN_LATENT_ENTROPY
 	default 1280 if (!64BIT && PARISC)
 	default 1024 if (!64BIT && !PARISC)
diff --git a/scripts/checksyscalls.sh b/scripts/checksyscalls.sh
index a18b47695f55..b3a6de55d18a 100755
--- a/scripts/checksyscalls.sh
+++ b/scripts/checksyscalls.sh
@@ -255,7 +255,7 @@ EOF
 syscall_list() {
     grep '^[0-9]' "$1" | sort -n |
 	while read nr abi name entry ; do
-		echo "#if !defined(__NR_${name}) && !defined(__IGNORE_${name})"
+		echo "#if 0"
 		echo "#warning syscall ${name} not implemented"
 		echo "#endif"
 	done
-- 
2.31.1.362.g311531c9de

